### ISSUE TITLE
[TEMP BUG FIXES] Working with Mapbox Unity SDK v2.1.1 - Location-based Game, Freezing, FPS spikes

### ISSUE THREAD

**Issue by Frixs** [mapbox/mapbox-unity-sdk#1886]
Hello,

I am currently working with Mapbox for a few weeks already.  
I am building simply-said location-based game and after a few interactions with Mapbox Unity SDK I found out, it's not perfect yet and there are many issues/bugs in the SDK.  

I can see, a lot of great work is made there already, but it needs a little bit more care to make it great overall.  
Anyway, after a few weeks of playing with the Mapbox Unity SDK, I ran into issues like FPS spikes, game/editor freezing, etc. I found solutions that I would like to share with you and hopefully they can help someone. I saw issues with FPS spikes already solved in the official dev, but not released in any new version. So this is like a temporary fix until the next version fixes it. :-)  
https://gist.github.com/Frixs/fa3b1bec1f7626c05c4c3c6ff1475618?permalink_comment_id=4337586#gistcomment-4337586

Hopefully, it can help you a bit and save some time in solving. :-)

---

**TL;DR**
How to solve FPS spikes /freezing? You might find help under the link above.
